### PR TITLE
[REFACTORING]  Improving PyGrid's code quality

### PR DESCRIPTION
**Pull request by IonesioJunior**
## Description
In order to improve the code quality, readability and maintainability, I am proposing a change in the way the source files are organized.  

**PS: The changes commented here do not change the behavior of the platform in any way. No endpoints or events had their API modified.**  

The ultimate goal of this PR is to improve the code quality to:  

- Improve code legibility by providing different abstraction levels, facilitating understanding by new contributors.
- Increase code modularization and reuse.
- Avoid monolithic codes and extensive classes.
- Reduce cyclomatic complexity.
- Define an intuitive and maintainable architecture.

To achieve this goal, the following modifications were made:

- The persistence models (previously defined in **app.main.storage.models**) were divided into different modules, having their own management entities. Thus, new modules were created.
- HTTP endpoints (previously defined in routes.py) were also divided by functionality, giving rise to a new module.

## Type of change

Please mark the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Code refactoring

## Checklist:

- [X] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
- [ ] New Unit tests added
- [ ] Unit tests pass locally with my changes

### CODE REVIEW

File: app/main/routes/federated.py [hunk 1]
```diff
@@ -418,17 +251,16 @@ def auth():
...
-        # @TODO: remove when hooked up to DB
-        # maybe processes._processes.last(name=model_name)["server_config"].get("pub_key", None)
+        pub_key = server.config.get("pub_key", None)
     else:
-        SECRET = "very long a$$ very secret key phrase"  #  TODO:@PRTFW remove after hookup to DB  # maybe processes._processes.last(name=model_name)["server_config"].get("JWT_SECRET", "very long a$$ very secret key phrase")
-
+        SECRET = server.config.get("JWT_SECRET", "very long a$$ very secret key phrase")
     """end stub DB vars"""
```
Comment (IonesioJunior) [inline app/main/routes/federated.py:264]: @Prtfw Could you verify/test if I fixed it properly?

Comment (wonderlabdevs) [reply app/main/routes/federated.py:264]: Yes. I will, feel free to merge and I can test that and submit issues for stuff that doesn't? if you prefer i test before merge, I can do that too : )

Comment (Prtfw) [reply app/main/routes/federated.py:264]: @IonesioJunior yes, can def do that, do you want to merge first and I can submit issues for the parts not working or do you want me to test first?

Comment (IonesioJunior) [reply app/main/routes/federated.py:264]: It's up to you! :)

Comment (Prtfw) [reply app/main/routes/federated.py:264]: If you are done / happy with the refactors, lets maybe merge this `big boy` first... I don't want to hold up the PR train : )
And we can start testing JWT as a separate issue. 
(I am starting to test but don't think the PR has to wait)

**Pull request merged**